packages/suite/src/support/suite/useConnectPopupWeb.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useRef, useState } from 'react';
+
+import {
+    CALL_SOURCE_WEB,
+    ManifestPartial,
+    connectPopupCallThunk,
+    connectPopupCancelThunk,
+    getPopupCallDeferred,
+    queuePopupCall,
+    selectConnectPopupCall,
+} from '@suite-common/connect-popup';
+import TrezorConnect, { IFRAME, POPUP, RESPONSE_EVENT, createPopupMessage } from '@trezor/connect';
+
+import { useDispatch, useSelector } from 'src/hooks/suite';
+
+const postMessageToParent = (message: any) => {
+    message.channel = {
+        here: '@trezor/connect-popup',
+        peer: '@trezor/connect-web',
+    };
+    if (window.opener) {
+        window.opener.postMessage(message, '*');
+    } else {
+        window.postMessage(message, window.location.origin);
+    }
+};
+
+export const useConnectPopupWeb = () => {
+    const dispatch = useDispatch();
+    const popupCall = useSelector(selectConnectPopupCall);
+    const lifecycle = useSelector(state => state.suite.lifecycle);
+    const manifest = useRef<ManifestPartial | undefined>(undefined);
+    const [pendingHandshake, setPendingHandshake] = useState<string | undefined>();
+
+    useEffect(() => {
+        const onMessage = async (event: MessageEvent) => {
```

Comment:
probably a nitpick - isn't it better to define onMessage outside of useEffect? This way I don't need to pass [dispatch] as its dependncy.